base/array.jl

```diff
@@ -2364,7 +2364,34 @@ function filter!(f, a::AbstractVector)
     return a
 end
 
-filter(f, a::Vector) = mapfilter(f, push!, a, empty(a))
+function filter!(f, a::Vector)
+    i = j = 1
+    L = length(a)
+    @inbounds while i <= L
```

Comment:
Is there any particular reason to write this as a while loop? `i` just goes over `1:L` and that is more easily read as a for-loop.
(Also avoids the overflow at `L=typemax(Int)`)

Reply:
No particular reason, no. I rewrote the loop for improved readability.